### PR TITLE
commit 06.04.2019 Serialization hw06

### PR DESCRIPTION
**Pull request by MateAcademy**

### CODE REVIEW

File: src/main/java/mate/academy/hw06/Animal.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+public class Animal implements Serializable {
+    private final String name;
+
+    public Animal(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw06/Animal.java:15]: why not override hash code also ?

File: src/main/java/mate/academy/hw06/Main.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        try (ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data))) {
+            int size = ois.readInt();
+            Animal[] deserializeAnimal = new Animal[size];
+
+            for (int i = 0; i < size; i++) {
+                deserializeAnimal[i] = (Animal) ois.readObject();
+            }
+            return deserializeAnimal;
+        } catch (ClassCastException | IOException | ClassNotFoundException e) {
+            throw new IllegalArgumentException();
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw06/Main.java:43]: print stack trace, e has information about error, when you throw new IllegalArgumentException(); you lose this information

**Review by Siril-O — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hw06/Animal.java (hunk 1 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw06/Animal.java:15]: this is the condition of task 5.4, the initial data https://stepik.org/lesson/Продвинутые-возможности-12785/step/8?thread=solutions&unit=3132

File: src/main/java/mate/academy/hw06/Main.java (hunk 2 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw06/Main.java:43]: This is condition of task 5.4, we have to throw an exception  IllegalArgumentException()

**Pull request merged**